sklearn/ensemble/forest.py

```diff
@@ -437,15 +437,16 @@ def _validate_y_class_weight(self, y):
         y = y_store_unique_indices
 
         if self.class_weight is not None:
-            valid_presets = ('auto', 'balanced', 'balanced_subsample', 'subsample', 'auto')
+            valid_presets = ('auto', 'balanced', 'subsample',
+                             'balanced_subsample')
             if isinstance(self.class_weight, six.string_types):
                 if self.class_weight not in valid_presets:
                     raise ValueError('Valid presets for class_weight include '
                                      '"balanced" and "balanced_subsample". Given "%s".'
                                      % self.class_weight)
                 if self.class_weight == "subsample":
-                    warn("class_weight='subsample' is deprecated and will be removed in 0.18."
-                         " It was replaced by class_weight='balanced_subsample' "
+                    warn("class_weight='subsample' is deprecated and will be removed in 0.19."
```

Comment:
"... is deprecated in 0.17 and will be removed in 0.19..."

It's important to be explicit with the version number when it was deprecated to make it easier to detect violations to our deprecation policy.